<launch>

  <arg name="scan_topic" default="/scan" />
  <arg name="start" default="0" />

  <!-- <env name="ROSCONSOLE_CONFIG_FILE"
    value="$(dirname)/../config/rosconsole.config" /> -->

  <!-- <param name="use_sim_time" value="true" /> -->

  <node pkg="gmapping" type="slam_gmapping_offline" name="slam_gmapping_offline"
    output="screen"
    args="-b /home/sousarbarb97/bag_files/paper_slam_2d/cappero_laser_odom_diag_2020-05-06-16-26-03_fixed_v2.bag
          --scantopic $(arg scan_topic)
          --start $(arg start)
          --log slam_gmapping.tum">
    <!-- launch-prefix="xterm -e gdb -/-args"/> -->

    <param name="base_frame" value="base_link" />
    <param name="odom_frame" value="odom" />
    <param name="map_frame" value="map" />

    <param name="map_update_interval" value="5.0" />

    <param name="delta" value="0.05" />
    <param name="occ_thresh" value="0.25" />
    <param name="throttle_scans" value="1" />

    <param name="maxUrange" value="8.0" />
    <param name="maxRange" value="10.0" />

    <param name="lskip" value="0" />
    <param name="sigma" value="0.05" />
    <param name="kernelSize" value="1" />
    <param name="lstep" value="0.05" />
    <param name="astep" value="0.05" />
    <param name="iterations" value="5" />
    <param name="lsigma" value="0.075" />
    <param name="ogain" value="3.0" />
    <param name="minimumScore" value="0.0" />

    <param name="srr" value="0.1" />
    <param name="srt" value="0.2" />
    <param name="str" value="0.1" />
    <param name="stt" value="0.2" />

    <param name="linearUpdate" value="1.00" />
    <param name="angularUpdate" value="0.5" />
    <param name="temporalUpdate" value="-1.0" />

    <param name="resampleThreshold" value="0.5" />
    <param name="particles" value="30" />

    <param name="xmin" value="-100.0" />
    <param name="ymin" value="-100.0" />
    <param name="xmax" value="100.0" />
    <param name="ymax" value="100.0" />

    <param name="llsamplerange" value="0.01" />
    <param name="llsamplestep" value="0.01" />
    <param name="lasamplerange" value="0.005" />
    <param name="lasamplestep" value="0.005" />

    <param name="transform_publish_period" value="0.05" />

  </node>

</launch>